File: src/main/java/io/vertx/ext/asyncsql/mysql/MysqlService.java
package io.vertx.ext.asyncsql.mysql;

import io.vertx.codegen.annotations.GenIgnore;
import io.vertx.codegen.annotations.ProxyGen;
import io.vertx.codegen.annotations.VertxGen;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.Vertx;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.asyncsql.BaseSqlService;
import io.vertx.ext.asyncsql.DatabaseCommands;
import io.vertx.ext.asyncsql.mysql.impl.MysqlServiceImpl;
import io.vertx.proxygen.ProxyHelper;

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>
 */
@VertxGen
@ProxyGen
public interface MysqlService extends BaseSqlService, DatabaseCommands {

  static MysqlService create(Vertx vertx, JsonObject config) {
    return new MysqlServiceImpl(vertx, config);
  }

  static MysqlService createEventBusProxy(Vertx vertx, String address) {
    return ProxyHelper.createProxy(MysqlService.class, vertx, address);
  }

  @Override
  void start(Handler<AsyncResult<Void>> whenDone);

  @Override
  void stop(Handler<AsyncResult<Void>> whenDone);

  /**
   * Begins a transaction.
   *
   * @return The transaction.
   */
  MysqlTransaction begin();

}

File: src/main/scala/io/vertx/ext/asyncsql/impl/MysqlService.scala
package io.vertx.ext.asyncsql.impl

import io.vertx.core.Vertx
import io.vertx.core.json.JsonObject
import io.vertx.ext.asyncsql.impl.pool.MysqlAsyncConnectionPool

/**
 * @author <a href="http://www.campudus.com">Joern Bernhardt</a>.
 */
class MysqlService(val vertx: Vertx, val config: JsonObject) extends BaseSqlService {

  override protected val poolFactory = MysqlAsyncConnectionPool.apply _

  override protected val defaultHost: String = "localhost"

  override protected val defaultPort: Int = 3306

  override protected val defaultDatabase: Option[String] = Some("testdb")

  override protected val defaultUser: String = "vertx"

  override protected val defaultPassword: Option[String] = Some("password")

}
